editoast/src/views/timetable.rs

```diff
@@ -476,19 +503,40 @@ impl FromStr for TrainId {
     type Err = &'static str;
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
-        if s.contains('#') {
-            let parts: Vec<&str> = s.split('#').collect();
-            if parts.len() != 2 {
-                return Err("Invalid PacedTrainOccurrenceId format");
+        if let Some((paced_train_id, index)) = s.split_once('#') {
+            let paced_train_id = paced_train_id
+                .parse::<i64>()
+                .map_err(|_| "Invalid train id")?;
+            let index = index.parse::<u64>().map_err(|_| "Invalid occurrence id")?;
+            Ok(TrainId::PacedTrainOccurrence {
+                paced_train_id,
+                occurrence_ref: PacedTrainOccurrenceRef::BaseOccurrence { index },
+            })
+        } else if s.contains('@') {
+            let parts: Vec<&str> = s.split('@').collect();
+            if parts.len() != 3 {
+                return Err("Invalid exception occurrence format");
             }
-
             let paced_train_id = parts[0].parse::<i64>().map_err(|_| "Invalid train id")?;
-            let index = parts[1]
-                .parse::<u64>()
-                .map_err(|_| "Invalid occurence id")?;
+            let exception_key = parts[1].to_string();
+            if exception_key.is_empty() {
+                return Err("Invalid exception key");
+            }
+            let maybe_index = Some(parts[2])
+                .filter(|i| !i.is_empty())
+                .and_then(|i| i.parse::<u64>().ok());
```

Comment:
I am not sure you want to ignore that parsing errors. Maybe you can add a test case for something like `zero@3@not-a-number` to see what happens but my guess is that it will not fail, but we probably want this case to fail.